builtin/logical/jwt/secret_tokens.go

```diff
@@ -0,0 +1,159 @@
+package jwt
+
+import (
+	"fmt"
+	"time"
+	"encoding/json"
+
+	jwt "github.com/dgrijalva/jwt-go"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+// SecretCertsType is the name used to identify this type
+const SecretTokensType = "token"
+
+func secretTokens(b *backend) *framework.Secret {
+	return &framework.Secret{
+		Type: SecretTokensType,
+		Fields: map[string]*framework.FieldSchema{
+			"token": &framework.FieldSchema{
+				Type: framework.TypeString,
+				Description: "The Signed JWT Token",
+			},
+		},
+
+		DefaultDuration: 720 * time.Hour,
+
+		Renew:  b.secretTokensRenew,
+		Revoke: b.secretTokensRevoke,
+	}
+}
+
+// Change FieldData for Map/Struct, so it is more versatile.
+
+func (b *backend) secretTokensCreate(
+	req *logical.Request, data *framework.FieldData, claims map[string]interface{}, roleName string) (*logical.Response, error) {
+
+	role, err := b.getRole(req.Storage, roleName)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return logical.ErrorResponse(fmt.Sprintf("Unknown role: %s", roleName)), nil
+	}
+
+	//token := jwt.New(jwt.SigningMethodRS256)
+	token := jwt.New(jwt.GetSigningMethod(role.Algorithm))
+
+	token.Claims = claims
+
+	tokenString, err := token.SignedString([]byte(role.Key))
+
+	if err != nil {
+		return nil, err
+	}
+
+	if role.Lease == true {
```

Comment:
I think this can be refactored a little. The "data" response should always be the same in either mode. The 'role' is not important to the data part, but is useful as internal data for renew.